cmd/cloud_sql_proxy/internal/healthcheck/healthcheck.go

```diff
@@ -33,12 +32,10 @@ const (
 
 // Server is a type used to implement health checks for the proxy.
 type Server struct {
-	// startedL protects started.
-	startedL sync.Mutex
-	// started is a flag that indicates whether the proxy is done starting up. 
-	// started is used to support readiness probing and should not be confused 
-	// for affecting startup probing.
-	started bool
+	// started is a channel used to indicate whether the proxy has finished 
+	// starting up. The channel is open when startup is in progress and
+	// closed when startup is complete.
+	started chan bool
```

Comment:
Fancy Go trick: if the value of the date on the channel doesn't matter (we never check for false or true), then `struct{}` is preferred.